src/formatter/string_formatter.rs

```diff
@@ -569,6 +569,23 @@ mod tests {
         match_next!(result_iter, "inner", inner_style);
     }
 
+    #[test]
+    fn test_style_variable_nested() {
+        const FORMAT_STR: &str = "[[text]($style)](blue)";
+        let inner_style = Some(Color::Red.bold());
+
+        let formatter = StringFormatter::new(FORMAT_STR)
+            .unwrap()
+            .map_style(|variable| match variable {
+                "style" => Some(Ok("red bold".to_owned())),
+                _ => None,
+            });
+
```

Comment:
The changes LGTM, but lets also test this:

```suggestion

        assert_eq!(
            formatter.style_variables.keys().collect::<Vec<_>>(),
            vec!["style"]
        );

```

Reply:
Done, but applied the same assertion using the public interface.